### DISCUSSION TITLE
windows 7

### DISCUSSION THREAD

**Discussion by Sandu119** [jovibor/Pepper#4] [General]
can it be run on windows 7?
It compiles ok in vs 2022 but I'm getting runtime errors about missing dll's.
I suspect it's more a matter of windows compatibility than of dll's actually missing

**Comment by jovibor** [comment]
I didn't test Windows 7 since it's too old/obsolete.
What exact errors you are getting?

**Reply by Sandu119** [reply]
Thank you for your answer.
Missing api-ms-win-core-version-l1-1-0.dll.
Strangely enough, Dependencies.exe doesn't list the above dll at all.

**Reply by jovibor** [reply]
> Missing api-ms-win-core-version-l1-1-0.dll.

This .dll is indeed used in the app.

![image](https://user-images.githubusercontent.com/5625787/194737316-9a850544-4c49-4d06-bc23-26220ed79a51.png)

I'm not sure about Windows 7, but maybe reinstalling the VS redist might help.
https://learn.microsoft.com/en-US/cpp/windows/latest-supported-vc-redist?view=msvc-170

Update:
According to [Microsoft ](https://learn.microsoft.com/en-us/windows/win32/api/winver/nf-winver-verqueryvaluew)this .dll is available since Windows 2000, so it should work under Windows 7 for sure.
![image](https://user-images.githubusercontent.com/5625787/194737670-685d9ced-bdfa-42c7-b679-a8a6d9a13045.png)

**Reply by Sandu119** [reply]
I already reinstalled the redist, no help.
Windows update did nothing.
I tried to manually install the dll, then I got a "missing api-ms-win-core-heap-l2-1-0.dll" error instead.
Installed the ..heap.. too and now I get "the procedure entry point LocalFree could not be located in api-ms-win-core-heap-l2-1-0.dll".
Couldn't find anywhere a api-ms-win-xxx.dll package.
The web is full of (mostly unresolved) questions about missing individual api-ms-win-...dll's
It seems strange though that VS 2022 successfully compiled the program without complaining about the missing dll's.
It seems hopeless.
